### DISCUSSION TITLE
How to set label and features for unlabeled node?

### DISCUSSION THREAD

**Discussion by JiaruiWang** [pyg-team/pytorch_geometric#4727] [Q&A]
My task is a node classification problem. The nodes in the graph are 40% labeled, and 60% unlabeled. 50 label classes. The label element in the Y tensor are 0~49 for regular classes, and -1 for unlabeled nodes conventionally.
I am using the ```GCNconv``` layer with ```add_self_loops=False```. The node features are the one-hot vectors for its label. There will be unlabeled nodes in the computation graph during training. I am not sure how to set the features for these unlabeled nodes since their class label is -1. I am thinking about two ways:
1. set the feature vector all 0 for unlabeled nodes.
2. extend the feature vector by 1 in length, to include -1 as a 'class'.

Which way is correct? Is there another better way?

**Comment by Padarn** [comment]
Are your labels input to your prediction? If so you might find the approach in https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/mask_label.py useful.

If they are not input maybe you should just mask these nodes while computing loss?

**Reply by JiaruiWang** [reply]
Yes, The label is the input. I want the GCNconv layer to predict the label for one node by looking at its neighbor's label.

**Reply by JiaruiWang** [reply]
Thank you! I found the answer in the paper. They used zero vectors for unlabeled nodes as I described in 1.

**Reply by Padarn** [reply]
great - good luck!
